parsley/shared/src/main/scala/parsley/internal/deepembedding/frontend/Visitors.scala

```diff
@@ -0,0 +1,267 @@
+/* SPDX-FileCopyrightText: © 2023 Parsley Contributors <https://github.com/j-mie6/Parsley/graphs/contributors>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.internal.deepembedding.frontend
+
+import parsley.debug.Breakpoint
+import parsley.errors.ErrorBuilder
+import parsley.internal.collection.immutable.Trie
+import parsley.internal.deepembedding.Sign.SignType
+import parsley.internal.deepembedding.singletons.*
+import parsley.internal.deepembedding.singletons.token.*
+import parsley.internal.errors.CaretWidth
+import parsley.registers.Reg
+import parsley.token.descriptions.SpaceDesc
+import parsley.token.descriptions.numeric.PlusSignPresence
+import parsley.token.errors.{ErrorConfig, LabelConfig, SpecialisedFilterConfig}
+import parsley.token.predicate.CharPredicate
+
+/** Visitor class template for the processing of parsers without fully explicit exhaustive pattern
+  * matching.
+  *
+  * This particular visitor is indexed on its return type to allow the preservation of relevant type
+  * information, such as setting it to [[LazyParsley]] to tell the type system that you want to
+  * produce new parsers using the visitor.
+  *
+  * @tparam T Context type for holding processing information as the visitor visits parsers.
+  * @tparam U Return value wrapper for the results of visiting the parsers.
+  */
+//noinspection ScalaStyle
+private [internal] abstract class LazyParsleyIVisitor[-T, +U[+_]] {
```

Comment:
`// scalastyle:ignore number.of.methods` at end of line

Reply:
remove the `//noinspection`

Reply:
added at end of both class declarations